### DISCUSSION TITLE
Training a vocoder with predicted mel spectrograms.

### DISCUSSION THREAD

**Discussion by francois-vz** [coqui-ai/TTS#2307] [General Q&A] [answered]
I have trained a Tacotron2 model with a custom phoneme set. Post-processing the output with a Univnet vocoder seems to significantly improve the performance of the synthesizer, so I would like to know whether it is possible to train the Univnet vocoder on both the wavs and the predicted mel spectograms of the Tacotron2 model?

```json
    "data_path": "some_path_to_wavs",
    "feature_path": null,
```
Is it possible to exploit the feature path config parameter to point the Univnet to the Tacotron2 spectograms of the training data? Also, how can I generate spectograms of the training data? Thanks for your trouble :)

**Comment by erogol** [comment, marked as answer]
There is ```TTS/bin/extract_tts_spectrograms.py``` to compute but I think it is outdated. you might need to update the code. 
Then you can add it to ```feature_path``` in the config and start training the vocoder.

**Comment by connershoop** [comment]
Hey @Fransie-dev did you have any luck here. I am doing the same thing and I can use `TTS/bin/extract_tts_spectrograms.py` with tacotron2_ddc_ph after removing [these lines](https://github.com/coqui-ai/TTS/blob/090cadf270711a61e3396f2f31eaaad54e32b5c1/TTS/bin/extract_tts_spectrograms.py#L44-L46)

**Reply by francois-vz** [reply]
Hi @connershoop, yes I did the same as you did :)
